Makefile

```diff
@@ -46,13 +45,35 @@ ALL_TEST_TAGS := $(ALL_BUILD_TAGS),$(TEST_TAG)
 BUILD_TAG_FLAG := -tags $(ALL_BUILD_TAGS)
 TEST_TAG_FLAG := -tags $(ALL_TEST_TAGS)
 
+# 20 minutes is the upper bound defined for all tests. (Tests in CI take up to about 12:30 now)
+# If you change this, also change .github/workflows/run-tests.yml!
+# The timeout in the GH workflow must be larger than this to avoid GH timing out the action,
+# which causes the a job run to not produce any logs and hurts the debugging experience.
+TEST_TIMEOUT ?= 20m
+
+# Number of retries for *-coverage targets.
+# NOTE: This is incompatible with TEST_ARGS which specify the `-run` flag due to how gotestsum selects which tests to
+# retry.
+FAILED_TEST_RETRIES ?= 2
+
+# Whether or not to test with the race detector. All of (1 on y yes t true) are true values.
+TEST_RACE_FLAG ?= on
+# Whether or not to shuffle tests. All of (1 on y yes t true) are true values.
+TEST_SHUFFLE_FLAG ?= on
+# Common test args used in the various test suite targets.
+COMPILED_TEST_ARGS := -timeout=$(TEST_TIMEOUT) \
```

Comment:
Hmm.. should we use TEST_TIMEOUT here? This is used for unit tests too and they have a different timeout in the GHA workflow...

Reply:
Ah, okay, I thought you were referring to a change I made.